### PR TITLE
create CSV download

### LINKED ISSUE #53: CSV download

### PR DESCRIPTION
**Pull request by dealingwith**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -149,18 +167,10 @@ def get_tasks
...
-    # else use all tasks
-    if (session[:parent_property_name])
-      tasks_no_subtasks = process_subtasks(tasks, config)
-    else
-      tasks_no_subtasks = tasks
-    end
-  # else, things have gone wrong
-  else
-    tasks_no_subtasks = []
+    tasks = process_subtasks(tasks, config) if process_subtasks
```
Comment (jamespeilunli) [inline app.rb]:
```suggestion
    tasks = process_subtasks(tasks, config) if process_subtasks?
```

File: app.rb [hunk 2]
```diff
@@ -121,7 +122,24 @@
...
+      title&.strip!
+      title = 'Untitled' if title.nil? || title == ''
+      csv << [title, "#{NOTION_BASE_URL}#{title.tr(" ", "-")}-#{task['id'].tr("-", "")}"]
+    end
+  end
+
+  send_file(temp_file.path, :filename => "tasksheet.csv")
+end
+
+def get_tasks(process_subtasks = true)
```
Comment (jamespeilunli) [inline app.rb]:
```suggestion
def get_tasks(process_subtasks? = true)
```

**Review by jamespeilunli — CHANGES_REQUESTED** [review]
`process_subtasks` is being used as a variable name and a method name on line 170. That's a bit confusing, so we should probably rename something.

**Comment by dealingwith** [comment]
vars ending with ? don't work in Ruby so I renamed the var entirely. Rebased and force-pushed a fresh single commit.

**Comment by jamespeilunli** [comment]
Didn't know that, thanks for letting me know haha

**Review by jamespeilunli — APPROVED** [review]

**Comment by dealingwith** [comment]
> Didn't know that, thanks for letting me know haha

I wasn't sure myself until I tested it. There are methods in Ruby and Rails with `?` suffixes when appropriate, e.g. `is_present?`, etc.

**Pull request merged**